react/REACT1/week3/todo-list-app/src/todoListComponent.js

```diff
@@ -0,0 +1,119 @@
+import React from "react";
+import { useState, useEffect } from "react";
+import TodoItem from "./todoitem";
+import BorderComponent from "./border";
+
+export function TodoList() {
+  const [todosState, setTodosState] = useState([]);
+  useEffect(() => {
+    fetch(
+      "https://gist.githubusercontent.com/benna100/391eee7a119b50bd2c5960ab51622532/raw"
+    )
+      .then((res) => res.json())
+      .then((data) => setTodosState(data));
+  }, []);
+  const current = new Date();
+  const todaysDate = `${current.getFullYear()}-0${current.getMonth() +
+    1}-${current.getDate()}`;
+
+  const [textInputState, setTextInput] = useState("");
+  const [dateInputState, setDateInput] = useState(todaysDate);
+
+  const addTodo = () => {
+    console.log(todaysDate, dateInputState);
+    if (
+      textInputState === "" ||
+      dateInputState === "" ||
+      dateInputState < todaysDate
+    ) {
+      alert(
+        "Can't add todo without correct description or deadline after todays date"
+      );
+    } else {
+      if (todosState.length === 0) {
+        const newTodo = {
+          description: textInputState,
+          deadline: dateInputState,
+          id: todosState.length + 1,
+        };
+        setTextInput("");
+        setDateInput("");
+        setTodosState((prevTodos) => {
+          return [...prevTodos, newTodo];
+        });
```

Comment:
Don't quite see the need for the callback function as an argument. Couldn't it just be like this?
`setTodosState([...todosState, newTodo])`